ament_tools/build_types/cmake.py

```diff
@@ -149,20 +173,24 @@ def _common_cmake_on_build(self, should_run_configure, context, prefix, extra_cm
                 if vsv not in supported_vsv:
                     raise VerbExecutionError('Unknown / unsupported VS version: ' + vsv)
                 cmake_args += ['-G', supported_vsv[vsv]]
+            elif IS_MACOSX:
+                if context.use_xcode or self._using_xcode_generator(context):
+                    cmake_args += ['-G', 'Xcode']
+                    cmake_args += ['-DCMAKE_XCODE_GENERATE_SCHEME=ON']
```

Comment:
I've removed the `CMAKE_XCODE_GENERATE_SCHEME` flag and used the `RUN_TESTS` target instead for `ament test` as a workaround in https://github.com/ament/ament_tools/pull/127/commits/eaf9430d79f3f1af4db1fd98c23cbffe53cd9180 This approach is compatible with CMake 3.5, so no need to use anything experimental from 3.8